src/helix.idea.vim

```diff
@@ -0,0 +1,189 @@
+"""""""""""""""""""""""""
+" IdeaVim support       "
+"""""""""""""""""""""""""
+set nohlsearch                    " no highlight search results
+set number                        " line numbers
+set showmode                      " show current mode
+set gdefault                      " substitute all occurrences in line per default
+set history=10000                 " keep x lines of command line history
+set scrolloff=5 sidescrolloff=10  " keep some lines before and after the cursor visible
+set visualbell                    " no beep
+set smartcase                     " no ignore case when pattern is uppercase
+set incsearch                     " show search results while typing
+set wrapscan                      " searches wrap around the end of the file
+
+set ideajoin
+set idearefactormode=keep
+
+" Plugins
+set argtextobj
+set matchit
+set multiple-cursors
+
+xnoremap \= <Action>(ReformatCode)
+
+" Select/extend
+nnoremap e ve
```

Comment:
Do the original helix.vim commands not work? I did some careful crafting on those to make such it would emulate the helix behavior for 99%. While I cannot test, I would imagine that these mappings are somewhat dissimilar to what helix does. Also, what if you press `e` twice, will that not cancel the selection because of the visual mode toggle?

Reply:
> Do the original helix.vim commands not work? I did some careful crafting on those to make such it would emulate the helix behavior for 99%.

I don't have the slightest clue what this actually does:
```vim
noremap e <Esc>maembviwovbviwvlmc`blmbhvl[`o`
```

... but it does indeed work perfectly on _Vim_. With IdeaVim sadly, this is how each movement behaves:

![recording-2024-10-04_21 47 18](https://github.com/user-attachments/assets/8d30240d-6fda-4ce8-8e7f-3807551c0211)

On first line, I repeat `e`, on second `w`, on third `b` and on fourth `B`.

> Also, what if you press e twice, will that not cancel the selection because of the visual mode toggle?

Yes I believe this is what happens. However, with these bindings, at least the first motion retains the selection. So they are more or less a workaround. The same motions with the included mappings:

![recording-2024-10-04_22 00 07](https://github.com/user-attachments/assets/c4044e94-a4ab-438c-a586-e16cfd786712)

Reply:
Is this good enough? My Vim-foo is not on the same level as yours. I don't have the slightest idea how to try to work around this...

Reply:
I see, that looks like IdeaVim has some offset regressions. I  Although ````noremap e <Esc>maembviwovbviwvlmc`blmbhvl[`o```` might seem intimidating, my vim skills also aren't that good :) It's mostly a bunch of mark commands and recalls that I figured out using trial and error & the help documentation. But I totally understand if you want to keep the simple version.

Reply:
I'd be stubborn enough to try fixing it, but I know for certain it'd take hours and hours. I'm hoping someone comes up who has enough Vim background to hack it together from muscle memory. :D